service/static/js/rest_api.js

```diff
@@ -73,7 +73,54 @@ $(function () {
     // ****************************************
     $('#createPromotionForm').on('submit', function (e) {
         e.preventDefault();  // Prevent default form submission behavior
-        // TODO : create a promotion
+
+        // $("#promotion_name").val(res.name);
+        // $("#promotion_product_id").val(res.product_id);
+        // $("#promotion_start_date").val(res.start_date);
+        // $("#promotion_end_date").val(res.end_date);
+        // $("#promotion_date_range_start").val(res.date_range_start);
+        // $("#promotion_date_range_end").val(res.date_range_end);
+        // $("#promotion_active_status").val(res.active_status.toString());
+        // $("#promotion_creator").val(res.creator);
+        // $("#promotion_updater").val(res.updater);
+
+        let id_prefix = "create_";
+        
+        let name = $(`#${id_prefix}promotion_name`).val();
+        let product_ids = $(`#${id_prefix}promotion_product_ids`).val();
+        let start_date = $(`#${id_prefix}promotion_start_date`).val();
+        let end_date = $(`#${id_prefix}promotion_end_date`).val();
+        let active_status = $(`#${id_prefix}promotion_active_status`).val() == "Active";
+        let creator = $(`#${id_prefix}promotion_creator`).val();
+        let updater = $(`#${id_prefix}promotion_updater`).val();
+        let extra = $(`#${id_prefix}promotion_extra`).val();
+
+        let data = {
+            name,
+            product_ids: product_ids.split(",").map(_ => _.trim()),
+            start_date,
+            end_date,
+            active_status,
+            created_by: creator,
+            updated_by: updater,
+            extra: JSON.parse(extra),
```

Comment:
extra should be an optional column. but if I don't put anything in this field it will get errors `Unexpected end of JSON input`

Reply:
I will add some non-empty string conditional branches here.